src/RelayClient.ts

```diff
@@ -456,40 +452,95 @@ class RelayClient extends EnvelopingEventEmitter {
     };
   }
 
+  private async _getRelayServer(): Promise<RelayInfo> {
+    if (this._relayServerUrl) {
+      return await this._getRelayInfo();
+    }
+
+    const { preferredRelays } = getEnvelopingConfig();
+
+    for (const preferredRelay of preferredRelays ?? []) {
+      try {
+        return await this._getRelayInfo(preferredRelay);
+      } catch (error) {
+        log.warn('Failed to getChainInfo from hub', error);
+        continue;
+      }
+    }
+
+    log.error('No more hubs available to select');
+
+    throw new Error('No more hubs available to select');
+  }
+
+  private async _getRelayInfo(relayServerUrl = this._relayServerUrl) {
+    try {
```

Comment:
I don't see any benefit of having this try/catch here.

Reply:
Right, changed.